src/Compilers/Core/Portable/Text/TextLine.cs

```diff
@@ -81,6 +81,14 @@ public static TextLine FromSpan(SourceText text, TextSpan span)
             }
         }
 
+        // Do not use unless you are certain the span you are passing in is valid!
+        // This was added to allow SourceText's indexer to directly create TextLines
+        // without the performance implications of calling FromSpan.
+        internal static TextLine FromKnownSpan(SourceText text, TextSpan span)
```

Comment:
Can we add an assert to verify the span in debug mode?

Reply:
Good idea, will do!

Reply:
Seems like a good idea, changed.